components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -647,6 +662,19 @@ esp_err_t esp_websocket_client_set_uri(esp_websocket_client_handle_t client, con
     return ESP_OK;
 }
 
+esp_err_t esp_websocket_client_set_headers(esp_websocket_client_handle_t client, const char *headers)
+{
+    if (client == NULL || headers == NULL) {
```

Comment:
Could we also check if the client is connected, or in some *invalid state* ? This would be ignored, correct? Or used in another connection attempt?